File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q32-concatenated-words.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var findAllConcatenatedWordsInADict = function(words) {
    const set = new Set(words);
    const result = [];
    
    for (let word of words) {
        if (dfs(word, 0, set, 0)) {
            result.push(word);
        }
    }
    
    return result;
    
};

function dfs(word, start, set, count) {
    if (start === word.length) {
        return count > 1;
    }
    
    for (let i = start + 1; i <= word.length; i++) {
        const prefix = word.substring(start, i);
        if (set.has(prefix)) {
            if (dfs(word, i, set, count + 1)) {
                return true;
            }
        }
    }
    
    return false;
}

File: ELEC881_Copilot/src/No Description Prompt/Python/q32-concatenated-words.py
class Solution(object):
    def findAllConcatenatedWordsInADict(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        words = set(words)
        def dfs(word):
            for i in range(1, len(word)):
                prefix, suffix = word[:i], word[i:]
                if prefix in words and suffix in words:
                    return True
                if prefix in words and dfs(suffix):
                    return True
                if suffix in words and dfs(prefix):
                    return True
            return False
        return [word for word in words if dfs(word)]